examples/example.hpp

```diff
@@ -497,32 +594,96 @@ class window
+            });
 
-        glfwSetScrollCallback(win, [](GLFWwindow * w, double xoffset, double yoffset)
-        {
-            auto s = (window*)glfwGetWindowUserPointer(w);
-            s->on_mouse_scroll(xoffset, yoffset);
-        });
+        glfwSetScrollCallback(win, [](GLFWwindow* w, double xoffset, double yoffset)
+            {
+                auto s = (window*)glfwGetWindowUserPointer(w);
+                s->on_mouse_scroll(xoffset, yoffset);
+            });
 
-        glfwSetCursorPosCallback(win, [](GLFWwindow * w, double x, double y)
-        {
-            auto s = (window*)glfwGetWindowUserPointer(w);
-            s->on_mouse_move(x, y);
-        });
+        glfwSetCursorPosCallback(win, [](GLFWwindow* w, double x, double y)
+            {
+                auto s = (window*)glfwGetWindowUserPointer(w);
+                s->on_mouse_move(x, y);
+            });
+
+        glfwSetKeyCallback(win, [](GLFWwindow* w, int key, int scancode, int action, int mods)
+            {
+                auto s = (window*)glfwGetWindowUserPointer(w);
+                if (0 == action) // on key release
+                {
+                    s->on_key_release(key);
+                }
+            });
+    }
 
-        glfwSetKeyCallback(win, [](GLFWwindow * w, int key, int scancode, int action, int mods)
+    //another c'tor for adjusting specific frames in specific tiles, this window is NOT resizeable
+    window(int width, int height, const char* title, int tiles_in_row, int tiles_in_col, float canvas_width = 0.8,
+        float canvas_height = 0.6, float canvas_left_top_x = 0.1, float canvas_left_top_y = 0.075)
+        : _width(width), _height(height), _tiles_in_row(tiles_in_row), _tiles_in_col(tiles_in_col)
+
+    {
+        //user input verification, if invalid values were given - set to default
+        if (canvas_width < 0 || canvas_width > 1 || canvas_height < 0 || canvas_height > 1 ||
+            canvas_left_top_x < 0 || canvas_left_top_x > 1 || canvas_left_top_y < 0 || canvas_left_top_y > 1)
         {
-            auto s = (window*)glfwGetWindowUserPointer(w);
-            if (0 == action) // on key release
+            std::cout << "Invalid window's size parameter entered, setting to default values" << std::endl;
+            canvas_width = 0.8;
+            canvas_height = 0.6;
+            canvas_left_top_x = 0.15;
+            canvas_left_top_y = 0.075;
+        }
+
+        //calculate canvas size
+        _canvas_width = _width * canvas_width;
+        _canvas_height = _height * canvas_height;
+        _canvas_left_top_x = _width * canvas_left_top_x;
+        _canvas_left_top_y = _height * canvas_left_top_y;
+
+        //calculate tile size
+        _tile_width_pixels = std::floor(_canvas_width / _tiles_in_row);
+        _tile_height_pixels = std::floor(_canvas_height / _tiles_in_col);
+
+        glfwInit();
+        // we don't want to enable resizing the window
+        glfwWindowHint(GLFW_RESIZABLE, GL_FALSE);
```

Comment:
To clarify - this class (and the constrain) is currently for HDR demo - it should not affect the other Demos.
Check that has no side-effects